nginx-stable fails to build from source with GCC 15 on amd64

The package nginx-stable with version 1.26.3-r42 fails to build from source with GCC 15 on amd64 with the following error:

```2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_sub_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_sub_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_addition_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_addition_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_gunzip_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_gunzip_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_userid_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_userid_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_headers_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_headers_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/ngx_http_copy_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/ngx_http_copy_filter_module.c
2025/05/31 00:19:05 WARN src/http/v2/ngx_http_v2_filter_module.c: In function 'ngx_http_v2_header_filter':
2025/05/31 00:19:05 WARN src/http/v2/ngx_http_v2_filter_module.c:118:36: error: initializer-string for array of 'unsigned char' truncates NUL terminator but destination lacks 'nonstring' attribute (6 chars into 5 available) [-Werror=unterminated-string-initialization]
2025/05/31 00:19:05 WARN   118 |     static const u_char nginx[5] = "��cU�";
2025/05/31 00:19:05 WARN       |                                    ^~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:19:05 WARN src/http/v2/ngx_http_v2_filter_module.c:121:9: error: initializer-string for array of 'unsigned char' truncates NUL terminator but destination lacks 'nonstring' attribute (13 chars into 12 available) [-Werror=unterminated-string-initialization]
2025/05/31 00:19:05 WARN   121 |         "���-i[D<��o";
2025/05/31 00:19:05 WARN       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_not_modified_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_not_modified_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_slice_filter_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_slice_filter_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/v2/ngx_http_v2.o \
2025/05/31 00:19:05 INFO     src/http/v2/ngx_http_v2.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/v2/ngx_http_v2_table.o \
2025/05/31 00:19:05 INFO     src/http/v2/ngx_http_v2_table.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/v2/ngx_http_v2_encode.o \
2025/05/31 00:19:05 INFO     src/http/v2/ngx_http_v2_encode.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/v2/ngx_http_v2_module.o \
2025/05/31 00:19:05 INFO     src/http/v2/ngx_http_v2_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_static_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_static_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_gzip_static_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_gzip_static_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_autoindex_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_autoindex_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_index_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_index_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_random_index_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_random_index_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_mirror_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_mirror_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_try_files_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_try_files_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_auth_request_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_auth_request_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_auth_basic_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_auth_basic_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_access_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_access_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_limit_conn_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_limit_conn_module.c
2025/05/31 00:19:05 WARN cc1: all warnings being treated as errors
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_limit_req_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_limit_req_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_realip_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_realip_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_geo_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_geo_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_map_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_map_module.c
2025/05/31 00:19:05 INFO cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I src/http/modules/perl -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 \
2025/05/31 00:19:05 INFO     -o objs/src/http/modules/ngx_http_split_clients_module.o \
2025/05/31 00:19:05 INFO     src/http/modules/ngx_http_split_clients_module.c
2025/05/31 00:19:06 WARN make[1]: *** [objs/Makefile:1120: objs/src/http/v2/ngx_http_v2_filter_module.o] Error 1
2025/05/31 00:19:06 WARN make[1]: *** Waiting for unfinished jobs....
2025/05/31 00:19:06 INFO make[1]: Leaving directory '/home/build/nginx-stable'
2025/05/31 00:19:06 WARN make: *** [Makefile:10: build] Error 2
2025/05/31 00:19:07 ERRO failed to build package: unable to run package nginx-stable pipeline: unable to run pipeline: exit status 2
make[1]: *** [Makefile:157: packages/x86_64/nginx-stable-1.26.3-r42.apk] Error 1
make[1]: Leaving directory '/home/sergio_durigan_chainguard_dev/os'
make: *** [Makefile:151: package/nginx-stable] Error 2
```

--- a/nginx-stable.yaml
+++ b/nginx-stable.yaml
@@ -1,7 +1,7 @@
 package:
   name: nginx-stable
   version: "1.27.5"
-  epoch: 0
+  epoch: 1
   description: HTTP and reverse proxy server (stable version)
   copyright:
     - license: BSD-2-Clause
@@ -24,7 +24,6 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - gd-dev
       - geoip-dev
       - libmaxminddb-dev
@@ -58,6 +57,10 @@ pipeline:
       tag: release-${{package.version}}
       expected-commit: 6ac8b69f06bc10d5503f636da888fa70095b151c
       destination: nginx-stable
+      # ref: https://github.com/nginx/nginx/pull/631
+      # TODO: remove once this get in stable release
+      cherry-picks: |
+        master/444954abacef1d77f3dc6e9b1878684c7e6fe5b3: Fixed -Wunterminated-string-initialization with gcc15.
   - name: configure
     working-directory: nginx-stable
     runs: |

